pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -208,6 +208,12 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 		if err != nil {
 			return reconcile.Result{}, err
 		}
+		if build.Spec.Runtime.Base.ImageURL != "" {
+			if err := AmendBuildStrategyWithRuntimeImage(buildStrategy, build); err != nil {
```

Comment:
I think we had many of us being in favor of using `generate_taskrun` instead. Have you considered it?

Reply:
Yes, the primary changes are in `generate_taskrun` now, please consider.